pkg/kubectl/cmd/create.go

```diff
@@ -132,7 +133,12 @@ func RunCreate(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *C
 			}
 		}
 
-		if err := createAndRefresh(info); err != nil {
+		client, err := f.DynamicClient(info.Mapping.GroupVersionKind.GroupVersion())
```

Comment:
I'm still not a fan of this.  I want my dynamic client to be able to accept `GroupVersionResources` dynamically, so that I can use a single dynamic client and say:
`dynamicClient.Post().Resource(GroupVersionResource).Body().Do()` and it will use its discovery based knowledge to construct the proper request URL.

HPA controller needs this, namespace lifecycle wants this, and ultimately, I think structures like this will benefit too.